service/src/test/java/com/sedliarov/learningtable/integration/TeacherControllerIntegrationTests.java

```diff
@@ -0,0 +1,206 @@
+  }
+
+  @Test
+  void testGetTeacherByIdIfUserNotExist() {
+    // when
+    ResponseEntity<Error> response =
+        exchangeGetWithoutAuth(TEACHERS_URL + TEACHER_UUID, Error.class);
+
+    // then
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.NOT_FOUND);
+    assertThat(response.getBody().getMessage()).isEqualTo(NOT_FOUND_PREFIX + TEACHER_WITH_ID
+        + TEACHER_UUID + NOT_FOUND);
+  }
+
+  @Test
+  void testGetTeachers() {
+    // given
+    Teacher teacher = teacherRepository.save(TeacherFixture
+        .createEntityWithFirstAndSecondName(FIRST_NAME_ARIA, SECOND_NAME_ARIEVNA));
+    Teacher teacher1 = teacherRepository.save(TeacherFixture
+        .createEntityWithFirstAndSecondName(FIRST_NAME_MARIA, SECOND_NAME_SHARAPOVA));
+    List<TeacherDto> expectedTeachers = List.of(mapper.entityToDto(teacher), mapper.entityToDto(teacher1));
+
+    // when
+    ResponseEntity<TeacherDto[]> response = exchangeGetWithoutAuth(TEACHERS_URL, TeacherDto[].class);
+
+    // then
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(response.getBody()).containsExactlyElementsOf(expectedTeachers);
+  }
+
+  @Test
+  void testGetTeachersIfNoUsers() {
+    // when
+    ResponseEntity<TeacherDto[]> response = exchangeGetWithoutAuth(TEACHERS_URL, TeacherDto[].class);
+
+    // then
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    assertThat(response.getBody()).containsExactlyElementsOf(Collections.emptyList());
+  }
+
+  @Test
+  void testCreateTeacher() {
+    // given
+    TeacherDto teacherForCreate = TeacherFixture.createDto();
+
+    // when
+    ResponseEntity<TeacherDto> response =
+        exchangePostWithoutAuth(TEACHERS_URL, teacherForCreate, TeacherDto.class);
+
+    // then
+    teacherForCreate.setTeacherId(response.getBody().getTeacherId());
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+    assertThat(response.getBody()).isEqualTo(teacherForCreate);
+  }
+
+  @Test
+  void testCreateTeacherIfExist() {
+    // given
+    Teacher teacherForSave = TeacherFixture.createEntity();
+    teacherRepository.save(teacherForSave);
+    TeacherDto expectedTeacher = mapper.entityToDto(teacherForSave);
+
+    // when
+    ResponseEntity<Error> response =
+        exchangePostWithoutAuth(TEACHERS_URL, expectedTeacher, Error.class);
+
+    // then
+    assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    assertThat(response.getBody().getMessage())
+        .isEqualTo(TEACHER + expectedTeacher.getFirstName() + " " + expectedTeacher.getSecondName()
+            + NOT_CREATED_BECAUSE_ALREADY_EXIST);
+  }
+
+  @Test
+  void testDeleteTeacher() {
+    // given
+    Teacher teacherForSave = TeacherFixture.createEntity();
+    teacherRepository.save(teacherForSave);
+
+    // when
+    ResponseEntity<TeacherDto> response =
+        exchangeDeleteWithoutAuth(TEACHERS_URL + teacherForSave.getTeacherId(), TeacherDto.class);
```

Comment:
To be sure that we won't have lazyException
```suggestion
    Teacher teacherForSave = TeacherFixture.createEntity();
    Teacher savedTeacher = teacherRepository.save(teacherForSave);

    // when
    ResponseEntity<TeacherDto> response =
        exchangeDeleteWithoutAuth(TEACHERS_URL + savedTeacher.getTeacherId(), TeacherDto.class);
```

Reply:
Done